### PR TITLE
fix(nvd): add support of `versionStartExcluding` and `versionEndExcluding`

### PR DESCRIPTION
**Pull request by DmitriyLewen**
## Description
- Add support of `versionStartExcluding` and `versionEndExcluding` fields.
- Add `(including)` and `(excluding)` suffixes for versions.
- Add `update` (if used) for versions from CPE

Before:
изображение

After:
изображение

### CODE REVIEW

File: docGen/nvd.go [hunk 1]
```diff
@@ -481,6 +480,22 @@ func parseVulnerabilityJSONFile(fileName string) (VulnerabilityPost, error) {
 	}, nil
 }
 
+func detectVersion(includeVersion, excludeVersion, itemVersion string) string {
```
Comment (DmitriyLewen) [inline docGen/nvd.go:483]: I used NVD logic (`(including)` and `(excluding)` suffixes).
Tell me if you see better way.

**Comment by DmitriyLewen** [comment]
@knqyf263 @simar7 Take a look this PR, when you have time, please.

File: docGen/nvd.go [hunk 2]
```diff
@@ -3,6 +3,7 @@ package main
 import (
 	"encoding/json"
 	"fmt"
+	"golang.org/x/exp/slices"
```
Comment (knqyf263) [inline docGen/nvd.go:6]: What if bumping Go to 1.21 or 1.22 and using `slices`?

Comment (DmitriyLewen) [reply docGen/nvd.go:6]: Why not.
Created #83 for Go update.

**Comment by pjonsson** [comment]
There is still a mismatch for the 0.9.0-beta versions in the nist.gov source and the screenshot in the PR, see the last screenshot from this comment: https://github.com/aquasecurity/vuln-list-update/issues/282#issue-2216503420

**Comment by DmitriyLewen** [comment]
Hello @pjonsson 
Thanks a lot! I missed `update` part for versions from `CPE`.

I updated this in https://github.com/aquasecurity/avd-generator/pull/82/commits/35c548c090c9a349996b356dd7ee859d7e3bc2e5

**Comment by pjonsson** [comment]
I don't know anything about the code, but the current screenshot in this PR looks like it captures everything from the CVE at nist.gov.

**Review by simar7 — APPROVED** [review]
nice! lgtm :shipit:

File: docGen/nvd.go (hunk 2 above)
Comment (simar7) [reply docGen/nvd.go:6]: @DmitriyLewen I merged this and the  other PRs so you can now use `slices`.

Comment (DmitriyLewen) [reply docGen/nvd.go:6]: Created #84

**Pull request merged**